nwb_conversion_tools/datainterfaces/ecephys/baserecordingextractorinterface.py

```diff
@@ -93,6 +93,10 @@ def subset_recording(self, stub_test: bool = False):
             kwargs.update(channel_ids=self.subset_channels)
 
         recording_extractor = se.SubRecordingExtractor(self.recording_extractor, **kwargs)
+
+        # Here, the gains need to be added again.
+        # But shis is specific, the code coulc check for neuroscope, and then call the function to add gains again.
+
```

Comment:
The problem is here. Here, some machinery to recover the gains that are eliminated by "subrecording" should be instantiated. Some ideas are:
1) call the methods in self .recording_extractor before to extract the gains (if available) and then re-assigned them at the end [if available]
2) call the method used in the `neuroscopedatainterface` to solve the problem. We can check if self.recording_extractor is the right one (the recorders have a name attribute) and then use the method.

Both seem kind *very* hack-y and I am not sure I feel comfortable with them. Moreover, will not generate the double scaling problem that @alejoe91 was mentioning in today (2022-01-04) meeting.